include/sparrow/mp_utils.hpp

```diff
@@ -192,7 +192,7 @@ namespace sparrow::mpl
     ///          or if the list is empty; `false` otherwise.
     template <class Predicate, template <class...> class L, class... T>
         requires any_typelist<L<T...>> and (callable_type_predicate<Predicate, T> && ...)
-    consteval bool all_of(L<T...> list, Predicate predicate)
+    consteval bool all_of([[maybe_unused]]L<T...> list, [[maybe_unused]] Predicate predicate)
```

Comment:
Note that it might be simpler here to just remove the name of `list`.
Also note that `predicate` will always be refered to, so it doesnt  need to be `maybe_unused`.

Reply:
I fixed the issue for list by just removing the name, but on GCC, the predicate parameter is detected as not used

Reply:
Looks like a bug? it's used in the return expression...
Maye add a "TODO: remove `[[mayb_unused]]` once gcc is fixed" in the comment you added there.